x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneSourceOperator.java

```diff
@@ -149,7 +148,7 @@ public Page getOutput() {
                 IntBlock leaf = null;
                 IntVector docs = null;
                 try {
-                    shard = blockFactory.newConstantIntBlockWith(scorer.shardIndex(), currentPagePos);
+                    shard = blockFactory.newConstantIntBlockWith(scorer.shardContext().index(), currentPagePos);
```

Comment:
thought: I was confused why using `.shardId()` is incorrect here, but the tests for `metadata _id` don't pass with this.

Reply:
Yeah - `shardId` is the count of the shard in the list of shards for the current index. Like, if you do `_cat/shards` it's the number that shows next to the shard. So, like, "this is the 0th shard of the index named foo". `index` is the index into the list. I'll see if i can make this more readable.